### PR TITLE
Ja Rule changes, Part 1

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
I'm trying to keep the PRs short and manageable.

File: plugins/usbdmx/LibUsbAdaptor.cpp [hunk 1]
```diff
@@ -180,7 +180,12 @@ int BaseLibUsbAdaptor::ClaimInterface(libusb_device_handle *dev,
 int BaseLibUsbAdaptor::DetachKernelDriver(libusb_device_handle *dev,
                                           int interface_number) {
   if (libusb_kernel_driver_active(dev, interface_number)) {
-    return libusb_detach_kernel_driver(dev, interface_number);
+    int r = libusb_detach_kernel_driver(dev, interface_number);
+    if (r) {
+      OLA_WARN << "libusb_detach_kernel_driver failed for: " << dev
+               << ": " << LibUsbAdaptor::ErrorCodeToString(r);;
```
Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: Remove the extra semicolon.

File: plugins/usbdmx/LibUsbAdaptor.cpp [hunk 2]
```diff
@@ -189,14 +194,35 @@ int BaseLibUsbAdaptor::DetachKernelDriver(libusb_device_handle *dev,
 int BaseLibUsbAdaptor::GetActiveConfigDescriptor(
       libusb_device *dev,
       struct libusb_config_descriptor **config) {
-  return libusb_get_active_config_descriptor(dev, config);
+  int r = libusb_get_active_config_descriptor(dev, config);
+  if (r) {
+    OLA_WARN << "libusb_get_active_config_descriptor failed for: " << dev
+             << ": " << LibUsbAdaptor::ErrorCodeToString(r);;
```
Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: Again.

File: plugins/usbdmx/LibUsbAdaptor.cpp [hunk 3]
```diff
@@ -189,14 +194,35 @@ int BaseLibUsbAdaptor::DetachKernelDriver(libusb_device_handle *dev,
...
+  return r;
+}
+
+int BaseLibUsbAdaptor::GetDeviceDescriptor(
+    libusb_device *dev,
+    struct libusb_device_descriptor *desc) {
+  int r = libusb_get_device_descriptor(dev, desc);
+  if (r) {
+    OLA_WARN << "libusb_get_device_descriptor failed for: " << dev
+             << ": " << LibUsbAdaptor::ErrorCodeToString(r);;
```
Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: Again.

File: plugins/usbdmx/LibUsbAdaptor.cpp [hunk 4]
```diff
@@ -189,14 +194,35 @@ int BaseLibUsbAdaptor::DetachKernelDriver(libusb_device_handle *dev,
...
 
 int BaseLibUsbAdaptor::GetConfigDescriptor(
     libusb_device *dev,
     uint8_t config_index,
     struct libusb_config_descriptor **config) {
-  return libusb_get_config_descriptor(dev, config_index, config);
+  int r = libusb_get_config_descriptor(dev, config_index, config);
+  if (r) {
+    OLA_WARN << "libusb_get_config_descriptor failed for: " << dev
+             << ": " << LibUsbAdaptor::ErrorCodeToString(r);;
```
Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: Again.

File: plugins/usbdmx/LibUsbAdaptor.h [hunk 5]
```diff
@@ -171,6 +181,16 @@ class LibUsbAdaptor {
   virtual void FreeConfigDescriptor(
       struct libusb_config_descriptor *config) = 0;
 
+  /**
+   * @brief Get the value of a string descriptor.
+   * @param usb_handle The USB device handle
+   * @param descriptor_index The index of the string descriptor to fetch.
+   * @param[out] data The value of the string descriptor.
+   */
```
Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.h]: @retuns?

**Comment by peternewman** [comment]
Generally LGTM, you could probably make the PR a little longer.

File: plugins/usbdmx/LibUsbAdaptor.h (hunk 5 above)
Comment (nomis52) [inline plugins/usbdmx/LibUsbAdaptor.h]: Done

File: plugins/usbdmx/LibUsbAdaptor.cpp (hunk 1 above)
Comment (nomis52) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: Changed all 4 to commas, we need something to separate the device from the error.

Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: I said semicolon, not colon. The one at the end of the line. Can you change the commas back to colons.

Comment (peternewman) [inline plugins/usbdmx/LibUsbAdaptor.cpp]: Fixed in a later PR.

**Comment by peternewman** [comment]
LGTM with later fixes.

**Pull request merged**